File: web/src/app/Models/Dtos/Test/test-result-dto.ts
import { DifficultyLevel } from '../../Enums/DifficultyLevel';
import { QuestionResultDto } from './question-result-dto';

export interface TestResultDto {
  testId: string;
  jobApplicationId: string;
  score: number;
  maxScore: number;
  passed: boolean;
  difficultyLevel: DifficultyLevel;
  questionResults: QuestionResultDto[];
}

File: backend/SkillProof/SkillProof.Entities/Dtos/Tests/TestResultDto.cs
using SkillProof.Entities.Enums;

namespace SkillProof.Entities.Dtos.Tests;

public class TestResultDto
{
    public string TestId { get; set; } = string.Empty;
    public string JobApplicationId { get; set; } = string.Empty;
    public double Score { get; set; }
    public int MaxScore { get; set; }
    public bool Passed { get; set; }
    public DifficultyLevel DifficultyLevel { get; set; }
    public List<QuestionResultDto> QuestionResults { get; set; } = new();
}
